synapse/storage/data_stores/main/room.py

```diff
@@ -448,21 +448,32 @@ def _get_media_mxcs_in_room_txn(self, txn, room_id):
         """
         mxc_re = re.compile("^mxc://([^/]+)/([^/#?]+)")
 
-        next_token = self.get_current_events_token() + 1
+        next_token = None
         local_media_mxcs = []
         remote_media_mxcs = []
 
-        while next_token:
-            sql = """
-                SELECT stream_ordering, json FROM events
-                JOIN event_json USING (room_id, event_id)
-                WHERE room_id = ?
-                    AND stream_ordering < ?
-                    AND contains_url = ? AND outlier = ?
-                ORDER BY stream_ordering DESC
-                LIMIT ?
-            """
-            txn.execute(sql, (room_id, next_token, True, False, 100))
+        while True:
+            if next_token is None:
+                sql = """
```

Comment:
suggestion: I think it would be clearer to move this outside the loop, and move the other case to the end (after the `break`), so that we always enter the loop with a populated txn.